src/pseudopeople/interface.py

```diff
@@ -102,12 +102,10 @@ def _generate_dataset(
 
 
 def validate_source_compatibility(source: Path):
-    # TODO: Clean this up w/ metadata
+    # TODO [MIC-4546]: Clean this up w/ metadata and update test_interface.py tests to be generic
     changelog = source / "CHANGELOG.rst"
     if changelog.exists():
-        with open(changelog, "r") as file:
-            first_line = file.readline()
-        version = parse(first_line.split("**")[1].split("-")[0].strip())
+        version = _get_data_changelog_version(changelog)
```

Comment:
I think we should make a metadata variable or something for the version instead of just having "1.4.2" as the parase arg.

Reply:
We will - that's what the TODOs and ticket MIC-4546 are all about